agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4440,32 +4440,41 @@ beerocks::message::sWifiChannel slave_thread::channel_selection_select_channel()
             continue;
         }
         for (uint8_t i = 0; i < beerocks::message::SUPPORTED_CHANNELS_LENGTH; i++) {
-            auto channel         = hostap_params.supported_channels[i];
```

Comment:
Why not store a reference to the channel if it's used in more than one place?

Reply:
actually now that you mention it, `const auto&` is indeed more appropriate. Just a reference without const is risky.